lib/finch/pool/selection_strategy/round_robin.ex

```diff
@@ -0,0 +1,19 @@
+defmodule Finch.Pool.SelectionStrategy.RoundRobin do
+  @moduledoc """
+  Cycle through the pools one by one, in a consistent order.
+  """
+  @behaviour Finch.Pool.SelectionStrategy
+
+  @impl true
+  def registry_value(%{count: count}) do
+    atomics = :atomics.new(1, [])
```

Comment:
I think you might want this to be unsigned. I think you're relying on overflow behavior and this is going into undefined behavior as it's eventually a signed-int overflow. I could be wrong.

Reply:
Great catch! I am relying on the overflow behaviour, but you're right that if the value becomes negative, things will break. Thank you!

Reply:
AFAIK `atomics` use intrinsics that guarantee proper wraparound for signed integers as well